src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -93,6 +97,115 @@ public ExerciseService(ExerciseRepository exerciseRepository, ExerciseUnitReposi
         this.exerciseUnitRepository = exerciseUnitRepository;
         this.programmingExerciseRepository = programmingExerciseRepository;
         this.quizExerciseRepository = quizExerciseRepository;
+        this.ltiOutcomeUrlRepository = ltiOutcomeUrlRepository;
+        this.studentParticipationRepository = studentParticipationRepository;
+    }
+
+    /**
+     * Gets the subset of given exercises that a user is allowed to see
+     *
+     * @param exercises exercises to filter
+     * @param user      user
+     * @return subset of the exercises that a user allowed to see
+     */
+    public Set<Exercise> getSubsetThatUserIsAllowedToSee(Set<Exercise> exercises, User user) {
+        if (exercises == null || user == null) {
+            throw new IllegalArgumentException();
```

Comment:
Not sure if this is good. Can you combine this with the if statement below and simply return an empty set?

Reply:
Done in 4e99b8ef26d26ac662dab4e3006674683c3fbcd2